From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Wed, 4 Feb 2026 21:24:55 +0100
Subject: gnome-desktop-thumbnail-script: Free template on error

g_mkdemp() may return error and in such case the passed templ must be
free'd

Origin: https://gitlab.gnome.org/GNOME/gnome-desktop/-/commit/0348462
---
 libgnome-desktop/gnome-desktop-thumbnail-script.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libgnome-desktop/gnome-desktop-thumbnail-script.c b/libgnome-desktop/gnome-desktop-thumbnail-script.c
index bbeddf0..80dada8 100644
--- a/libgnome-desktop/gnome-desktop-thumbnail-script.c
+++ b/libgnome-desktop/gnome-desktop-thumbnail-script.c
@@ -979,6 +979,7 @@ script_exec_new (const char  *uri,
       exec->outdir = g_mkdtemp (tmpl);
       if (!exec->outdir)
         {
+          g_clear_pointer (&tmpl, g_free);
           g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                                "Could not create temporary sandbox directory");
           goto bail;
@@ -1030,6 +1031,7 @@ script_exec_new (const char  *uri,
       exec->outdir = g_mkdtemp (tmpl);
       if (!exec->outdir)
         {
+          g_clear_pointer (&tmpl, g_free);
           g_set_error_literal (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                                "Could not create temporary sandbox directory");
           goto bail;
